### ISSUE TITLE
False Positive: Enum.__members__ 

### ISSUE THREAD

**Issue by bhargavrpatel** [PyCQA/pylint#3615]
### Steps to reproduce
1. Create a class like this:

```python
import typing
from enum import Enum


class MyEnum(Enum):
    def to_pb2(self) -> int:
        return typing.cast(int, self.value)

    def __str__(self) -> str:
        return self.name.upper()  # pylint: disable=no-member


class SomeEnum(MyEnum):
    A = "A"


class AnotherEnum(Enum):
    B = "B"


SomeEnum.__members__.items()
AnotherEnum.__members__.items()

```

2. Install pylint 2.5.4 and run: `pylint --errors-only sample_file.py`
3. Install pylint 2.5.2 and run: `pylint --errors-only sample_file.py`

### Current behavior
pylint 2.5.2 raises a no member warning

```
sample_file.py:21:0: E1101: Function '__members__' has no 'items' member (no-member)
```

### Expected behavior
This should not happen

### pylint --version output

```
$ pylint --version

pylint 2.5.2
astroid 2.4.1
Python 3.7.2 (default, Feb  2 2019, 01:42:52) 
[GCC 5.4.0 20160609]
```

**Comment by bhargavrpatel** [comment]
Closing: Dupe of https://github.com/PyCQA/pylint/issues/3615
